main/src/main/java/cgeo/geocaching/activity/AbstractBottomNavigationActivity.java

```diff
@@ -129,14 +130,18 @@ public View getView(final int position, final View convertView, @NonNull final V
                         GeoItemSelectorUtils.getOrCreateView(AbstractBottomNavigationActivity.this, convertView, parent));
             }
         };
-        Dialogs.newBuilder(this)
+        final AlertDialog.Builder builder = Dialogs.newBuilder(this)
                 .setTitle(R.string.cache_recently_viewed)
                 .setAdapter(adapter, (dialog, which) -> CacheDetailActivity.startActivity(this, lastCaches.get(which).getGeocode()))
                 .setPositiveButton(R.string.map_as_list, (d, w) -> {
                     CacheListActivity.startActivityLastViewed(this, new SearchResult(lastCaches));
                     ActivityMixin.overrideTransitionToFade(this);
                 })
-                .show();
+                .setNegativeButton(R.string.cache_clear_recently_viewed, (d, w) -> Settings.clearRecentlyViewedHistory());
```

Comment:
When there are no recently view caches, "clear" and "show as list" do not really make sense. Maybe just "ok" to close the dialog?

Reply:
Fair enough. When thinking about it, a dialog isn't necessary at all in this case. Will change it to a toast notification...